multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ValidateDeployParametersStep.java

```diff
@@ -114,10 +118,27 @@ private void checkFileSizeOfAllFiles(ProcessContext context) throws FileStorageE
         long maxFileSizeLimit = configuration.getMaxUploadSize();
         List<FileEntry> fileEntries = fileService.listFilesBySpaceAndOperationId(context.getVariable(Variables.SPACE_GUID),
                                                                                  context.getVariable(Variables.CORRELATION_ID));
-        long sizeOfAllFiles = getSizeOfAllFiles(fileEntries);
-        if (sizeOfAllFiles >= maxFileSizeLimit) {
-            deleteFiles(context, fileEntries);
-            throw new ContentException(Messages.SIZE_OF_ALL_OPERATIONS_FILES_0_EXCEEDS_MAX_UPLOAD_SIZE_1, sizeOfAllFiles, maxFileSizeLimit);
+        if (renameLater.shouldSumAllTheFiles(context.getExecution())) {
+            long sizeOfAllFiles = getSizeOfAllFiles(fileEntries);
+            if (sizeOfAllFiles >= maxFileSizeLimit) {
+                deleteFiles(context, fileEntries);
+                throw new ContentException(Messages.SIZE_OF_ALL_OPERATIONS_FILES_0_EXCEEDS_MAX_UPLOAD_SIZE_1,
+                                           sizeOfAllFiles,
+                                           maxFileSizeLimit);
+            }
+        } else {
+            for (FileEntry fileEntry: fileEntries) {
+                checkFileSizeOfAFile(maxFileSizeLimit, context, fileEntry);
```

Comment:
what will happen if the multiple uploaded archives are chunks?

Reply:
This check won't be fine in long term because there are plans TMS to swtich to chunk upload and then this validation will be odd and again it will be possible to upload bigger than 4gb mtar files.